lib/broadside/ecs/ecs_manager.rb

```diff
@@ -44,24 +44,43 @@ def get_latest_task_definition_arn(name)
         get_task_definition_arns(name).last
       end
 
+      def get_running_instance_ips!(cluster, family, task_arns = nil)
+        ips = get_running_instance_ips(cluster, family, task_arns)
+        raise Error, "No running tasks found for '#{family}' on cluster '#{cluster}'!" if ips.empty?
+        ips
+      end
+
       def get_running_instance_ips(cluster, family, task_arns = nil)
         task_arns = task_arns ? Array.wrap(task_arns) : get_task_arns(cluster, family)
-        raise Error, "No running tasks found for '#{family}' on cluster '#{cluster}'!" if task_arns.empty?
 
-        tasks = ecs.describe_tasks(cluster: cluster, tasks: task_arns).tasks
-        container_instances = ecs.describe_container_instances(
-          cluster: cluster,
-          container_instances: tasks.map(&:container_instance_arn),
-        ).container_instances
+        if task_arns.empty?
+          []
+        else
+          tasks = ecs.describe_tasks(cluster: cluster, tasks: task_arns).tasks
+          container_instances = ecs.describe_container_instances(
+            cluster: cluster,
+            container_instances: tasks.map(&:container_instance_arn),
+          ).container_instances
 
-        ec2_instance_ids = container_instances.map(&:ec2_instance_id)
-        reservations = ec2_client.describe_instances(instance_ids: ec2_instance_ids).reservations
+          ec2_instance_ids = container_instances.map(&:ec2_instance_id)
+          reservations = ec2_client.describe_instances(instance_ids: ec2_instance_ids).reservations
 
-        reservations.map(&:instances).flatten.map(&:private_ip_address)
+          reservations.map(&:instances).flatten.map(&:private_ip_address)
+        end
       end
 
-      def get_task_arns(cluster, family)
-        all_results(:list_tasks, :task_arns, { cluster: cluster, family: family })
+      def get_task_arns(cluster, family, filter = {})
+        opts = {
+          cluster: cluster,
+          family: family,
```

Comment:
ditch the line `opts[:family] = nil if opts[:service_name]` below and just go with
```
family: filter[:service_name] ? nil : family
```